Libraries/Components/WebView/WebView.ios.js

```diff
@@ -462,6 +462,38 @@ var WebView = React.createClass({
   },
 
   /**
+   * Evaluate JavaScript on the current page.
+   */
+  evaluateJavaScript: function(
+    script: string,
+    callback?: ?(error: ?Error, result: ?string) => void,
+  ): Promise {
+    return new Promise((resolve, reject) => {
+      var escaped = JSON.stringify(script).replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
+      var wrapped = 'JSON.stringify(eval(' + escaped + '))';
+
+      RCTWebViewManager.evaluateJavaScript(this.getWebViewHandle(), wrapped, function(error, result) {
+        if (!error) {
+          if (result === '') {
+            error = new Error("Error evaluating script.")
```

Comment:
@thebnich You can always check for `nil` on the native side and send an error message and error code to JS, something like e`Failed to evaluate JavaScript`, `ERR_JS_EVALUTION_FAILED`.

Checking for empty string here seems problematic. What if the result is empty string?